src/lib/pickles/impls.ml

```diff
@@ -36,11 +40,18 @@ module Step = struct
       Field.t * Boolean.var
 
     let forbidden_shifted_values =
-      forbidden_shifted_values ~size_in_bits:Constant.size_in_bits
-        ~modulus:(Wrap_impl.Bigint.to_bignum_bigint Constant.size) ~f:(fun x ->
-          let hi = test_bit x (Field.size_in_bits - 1) in
-          let lo = B.shift_right x 1 in
-          (Impl.Bigint.(to_field (of_bignum_bigint lo)), hi))
+      let f x =
+        let hi = test_bit x (Field.size_in_bits - 1) in
+        let lo = B.shift_right x 1 in
+        (Impl.Bigint.(to_field (of_bignum_bigint lo)), hi)
```

Comment:
This is subtly incorrect: to preserve the previous behaviour, this should set the field element to `0` if it is `>= Impl.Field.Constant.size`, rather than checking against the other field's modulus.

Reply:
do you know how come the CI tests are passing :o ?

Reply:
I need to tweak the script, no proofs are being accepted if you look [here](https://buildkite.com/o-1-labs-2/mina/builds/18489#659b2d10-a389-4c59-91e1-f43230db57f6)

Reply:
Ah indeed! Thanks, and good catch :) now I understand how to check if my fix works